CharlieBackend.Api/Validators/ScheduledEventDTOValidators/UpdateScheduledEventDTOValidator.cs

```diff
@@ -0,0 +1,73 @@
+﻿using CharlieBackend.Business.Exceptions;
+using CharlieBackend.Business.Helpers;
+using CharlieBackend.Core.DTO.Schedule;
+using CharlieBackend.Data.Repositories.Impl.Interfaces;
+using FluentValidation;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace CharlieBackend.Api.Validators.ScheduledEventDTOValidators
+{
+    public class UpdateScheduledEventDTOValidator : AbstractValidator<UpdateScheduledEventDto>
+    {
+        private readonly IUnitOfWork _unitOfWork;
+        
+        public UpdateScheduledEventDTOValidator(IUnitOfWork unitOfWork)
+        {
+            _unitOfWork = unitOfWork;
+
+            RuleFor(x => x.StudentGroupId)
+                            .MustAsync(async (StudentGroupId, cancellation)
+                    => await IsStudentGroupExistAsync(StudentGroupId.GetValueOrDefault()))
+                .When(x => x.StudentGroupId != null)
+                .OnAnyFailure(x =>
+                {
+                    throw new EntityValidationException(ValidationConstants.StudentGroupNotValid);
```

Comment:
Don't throw exceptions. Fluent validation throws it automatically.

Reply:
Added checking "WithMessage", deleted requests to the database